pkg/booleanpolicy/validate.go

```diff
@@ -152,3 +170,23 @@ func validatePolicySection(s *storage.PolicySection, configuration *validateConf
 
 	return errorList.ToError()
 }
+
+// validateFieldDependencies validates a policy section with respect to required
+// combinations of fields, as outlined in the fieldDependencies map.
+func validateFieldDependencies(s *storage.PolicySection, seenFields *set.StringSet) error {
+	errorList := errorhelpers.NewErrorList(fmt.Sprintf("validation field dependencies for %q", s.GetSectionName()))
+	for _, field := range seenFields.AsSlice() {
+		dependencies, found := fieldDependencies[field]
+		if !found {
+			continue
+		}
+
+		if !slices.ContainsFunc(dependencies.AsSlice(), func(dep string) bool {
```

Comment:
I would flip for faster iteration over fewer items. for each key in the dependencies map, if seenFields.Contains (key) and does not contain any of the dependencies, error.